### PR TITLE
add rocketmq-client metadata

### PR DESCRIPTION
**Pull request by yuluo-yx**
## What does this PR do?

- For https://github.com/oracle/graalvm-reachability-metadata/issues/260
 
## Checklist before merging
- [X] I have considered including reachability metadata directly in the library or the framework (see [this](./CONTRIBUTING.md))
- [X] I have properly formatted metadata files (see [this](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#format-metadata-files))
- [X] I have added thorough tests (see [this](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))

### CODE REVIEW

File: metadata/org.apache.rocketmq/rocketmq-client/4.9.4/reflect-config.json [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+        "name": "read",
+        "parameterTypes": [
+          "io.netty.channel.ChannelHandlerContext"
+        ]
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "org.apache.rocketmq.remoting.netty.NettyRemotingClient$4"
```
Comment (vjovanov) [inline metadata/org.apache.rocketmq/rocketmq-client/4.9.4/reflect-config.json:141]: Is this agent-generated or manual?

**Comment by vjovanov** [comment]
Thanks greatly for your contribution. Have you tried opening a ticket on the library itself to support native-image?

File: metadata/org.apache.rocketmq/rocketmq-client/4.9.4/reflect-config.json (hunk 1 above)
Comment (yuluo-yx) [reply metadata/org.apache.rocketmq/rocketmq-client/4.9.4/reflect-config.json:141]: > Is this agent-generated or manual?

It comes from agent generation.

**Comment by yuluo-yx** [comment]
> Thanks greatly for your contribution. Have you tried opening a ticket on the library itself to support native-image?



> opening a ticket on the library itself to support native-image

Sorry, I didn't understand what you mean by `opening a ticket on the library itself to support native-image`. Can you explain in detail?

**Comment by vjovanov** [comment]
Well go to the issue tracker of `rocketmq`  and ask them for support for Native Image.

**Comment by dnestoro** [comment]
@yuluo-yx Any update regarding this PR? Have you tried what @vjovanov suggested in the previous comment?

**Comment by yuluo-yx** [comment]
> @yuluo-yx Any update regarding this PR? Have you tried what @vjovanov suggested in the previous comment?

Thank you very much for continuing to follow this PR, now without any updates I will close it. @vjovanov

**Pull request closed without merging**